tests/6.0/parallel/test_parallel_safesync.c

```diff
@@ -0,0 +1,64 @@
+//--------------- test_parallel_safesync.c------------------------------------//
+// OpenMP API Version 6.0 November 2024
+// Pg. 901, line 9
+// ***********
+// DIRECTIVE:parallel
+// CLAUSE:safesync
+// ***********
+// Code is adapted from the safesync clause example in the 6.0 examples
+// document. This example uses a ticket lock implementation which executes in
+// the target region. Certain offloading implementations may be unable to
+// properly synchronize the threads pertaining to different progress groups at
+// the end of the parallel region. The result of this can be deadlock. The
+// safesync clause is suppossed to ensure that such synchronization is possible.
+// If no deadlock or inporper update of the arrays occurs, then this indicates
+// the safesync clause was working correctly.
+//----------------------------------------------------------------------------//
+#include "ompvv.h"
+#include <omp.h>
+
+#define N 100
+
+int test_directive() {
+  int errors = 0;
+  int a[N], b[N];
+  int count1 = 0, count2 = 0;
```

Comment:
In line 25, `count2 = 0;` should be changed to `count2 = 1;`. With `count2 = 0;` no thread can exit from the first do-while loop in line 37 - 40. The first thread entering the do-while loop will have `t = 1` and `u = 0` and thus cannot exit (`while (0 < 1)`), and all the later threads will be stuck in the same loop (`while(0<2)`, `while(0<3)`, ...). With `count2 = 1;` threads will exit from the first do-while loop one by one (`while(1<1)`, `while(2<2)`, …) (whenever previous thread increases `count2` in line 46).
(I think that the original example in the 6.0 example document has an error.)